model/src/channel/news_thread.rs

```diff
@@ -0,0 +1,141 @@
+use crate::channel::{ChannelType, ThreadMember, ThreadMetadata};
+use crate::id::{ChannelId, GuildId, MessageId, UserId};
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Debug, Deserialize, Eq, Hash, PartialEq, Serialize)]
+pub struct NewsThread {
+    pub id: ChannelId,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub guild_id: Option<GuildId>,
+    #[serde(rename = "type")]
+    pub kind: ChannelType,
+    pub name: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub owner_id: Option<UserId>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub parent_id: Option<ChannelId>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub last_message_id: Option<MessageId>,
+    pub message_count: u8,
```

Comment:
A u8 might be surprising to someone reading the documentation for this, so we should document why.